### ISSUE TITLE
Improve ifthenelse and Image docs

### ISSUE THREAD

**Issue by int-ua** [libvips/pyvips#203]
**Is your feature request related to a problem? Please describe.**
[`pyvips.Image.ifthenelse`](https://libvips.github.io/pyvips/vimage.html#pyvips.Image.ifthenelse) documentation is confusing.

**Describe the solution you'd like**
Make the documentation answer these questions:
 * What do `th` and `el` stand for? Readability counts.
 * If they are `then` and `else`, where is the `if`? After reading the second time I got that it's `self` which is named `cond` but it looks a bit inconsistent to give `th` and `el` new names (`in1` and `in2`) right after they were first introduced.
 * Why it says that it requires an image when all over the search results you get examples with lists and even sets in place of images? The [`Image`](https://libvips.github.io/pyvips/vimage.html#pyvips.Image) docs desperately lack an explanation for this. What do the numbers in these lists and sets stand for?

**Describe alternatives you've considered**
Sticking with my ImageMagick script that I had much less problems making.

**Comment by jcupitt** [comment]
You're right, th and el are a bit confusing. I changed them to in1 and in2 to match the introspection names. 

https://github.com/libvips/pyvips/commit/c9da36c4c5800051d239fc249ff90d8354660357

**Comment by jcupitt** [comment]
Did you find the main part of the python docs? It has a long explanation of the numbers and arrays:

https://libvips.github.io/pyvips/intro.html#calling-libvips-operations

The docs for member functions like `ifthenelse` etc. are generated automatically from introspection of the library and are not very chatty, unfortunately. The main docs for ifthenelse have more detail:

https://libvips.github.io/libvips/API/current/libvips-conversion.html#vips-ifthenelse

**Comment by int-ua** [comment]
Reading the docs, do you mind if I post my notes here? I'll start.

https://libvips.github.io/pyvips/intro.html

> boosts the green channel, sharpens the image, and saves it

Maybe these will help more as comments in the code to better point to what does what

>The default mode is random

What are the benefits of that?

> The scale is the amount to divide the image by after integer convolution.

Can you please add a simplified explanation after or before this?

`conv` being convolution was unexpected, in the code I first read it as "convert" and thus it was confusing.

Maybe words like "combine" and/or "mix" can be used in a simplified explanation.

https://libvips.github.io/pyvips/vimage.html#pyvips.Image.conv

> Convolution operation.

What does that mean in the context of images and in simpler terms? Or, at least, what can it be used for?

Reading further.

Although it may be excessive and hard to do, may I suggest asking someone who is close to you but less familiar with image processing yet read these docs and ask you what is not clear with writing down the questions and answers in the process to use these notes for improvements?

**Comment by int-ua** [comment]
> There are the usual range of arithmetic operator overloads.

I can guess them, but it wouldn't hurt listing at least some examples or linking to a page with them.

> ipct_string = image.get('ipct-data')

Wouldn't hurt linking to an explanation that educates on differences with EXIF, I guess. Maybe as an inline comment?

> behind your back

has negative connotations IMHO, how about "behind the scenes"? But it's really insignificant, ignore if you have any doubts.

> so you can’t change an image unless you are certain you have the only reference to it.

I _guess_ the next paragraphs are the explanation but the connection is not obvious. There is something implied here that should be said explicitly, although I'm not sure what.

**Comment by int-ua** [comment]
> Any trailing keyword arguments are used to set options on the operation.

What is an option in this context? An optional argument?

> The result is the required output argument if there is only one result

How about something along the lines of "The returned value can be only one object (I guess it's not always an Image) or a tuple? list? https://docs.python.org/3/library/array.html ? of objects for some operations."

> If the operation has optional output objects, they are returned as a final hash.

Do you mean "dictionary"?
How about "The final object in the list/tuple/array can be a dictionary for operations with optional output objects."?

**Comment by jcupitt** [comment]
Thank you for the suggestions. It's interesting to see an outsider's take -- I've been looking at this stuff for so long that of course it all seems simple and obvious to me.

I've tried to incorporate your ideas and credited you in CHANGES.

**Comment by int-ua** [comment]
Thank you. This tool look like having a great potential but inaccessible at times without a heavier background in image processing.
